File: src/main/java/io/github/oleksivio/telegram/bot/api/annotations/filter/telegram/EncryptedCredentialsFilter.java
package io.github.oleksivio.telegram.bot.api.annotations.filter.telegram;

import io.github.oleksivio.telegram.bot.api.annotations.filter.primitive.NotNullFilter;
import io.github.oleksivio.telegram.bot.api.model.annotation.AnnotationState;

import java.lang.annotation.Documented;
import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.METHOD)
@Documented
public @interface EncryptedCredentialsFilter {

    AnnotationState status() default AnnotationState.ON;

    String[] validator() default {};

    NotNullFilter data() default @NotNullFilter(status = AnnotationState.OFF);

    NotNullFilter hash() default @NotNullFilter(status = AnnotationState.OFF);

    NotNullFilter secret() default @NotNullFilter(status = AnnotationState.OFF);

}

File: src/main/kotlin/io/github/oleksivio/telegram/bot/api/annotations/filter/composite/EncryptedCredentialsFilter.kt
package io.github.oleksivio.telegram.bot.api.annotations.filter.composite

import io.github.oleksivio.telegram.bot.api.annotations.filter.primitive.NotNullFilter
import io.github.oleksivio.telegram.bot.api.model.annotation.AnnotationState

@Retention
@Target(AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
@MustBeDocumented
annotation class EncryptedCredentialsFilter(val status: AnnotationState = AnnotationState.ON, val validator: Array<String> = [], val data: NotNullFilter = NotNullFilter(status = AnnotationState.OFF), val hash: NotNullFilter = NotNullFilter(status = AnnotationState.OFF), val secret: NotNullFilter = NotNullFilter(status = AnnotationState.OFF))
